action.yml

```diff
@@ -46,13 +46,41 @@ runs:
             const diffInSeconds = (currentRunDurationInMillis - latestMasterRunDurationInMillis) / 1000
             const percentageDiff = ((1 - (currentRunDurationInMillis/latestMasterRunDurationInMillis)) * 100).toFixed(2)
             const outcome = (diffInSeconds > 0) ? "an increase" : "a decrease"
-         
+
             outputMessage =  '🕒 Workflow \"' + context.workflow + '\" took ' +  (currentRunDurationInMillis / 1000) + 's which is ' + outcome + ' with ' + Math.abs(diffInSeconds) + 's (' + Math.abs(percentageDiff) + '%) compared to latest run on master/main.'
           }
 
-          github.rest.issues.createComment({
-            issue_number: context.issue.number,
+          var commentData = {
             owner: context.repo.owner,
             repo: context.repo.repo,
-            body: outputMessage
+            issue_number: context.issue.number,
+          };
+
+          var comments = github.rest.issues.listComments(commentData);
+
+          /**
+          * Add the body content after comments are fetched so that it's
+          * not included in the search for existing comments.
+          */
+          commentData.body = outputMessage;
+
+          /**
+          * Search comments from the bottom-up to find the most recent comment
+          * from the GitHub Actions bot that matches our criteria.
+          */
+          var existingComment = comments.reverse().find(comment => {
```

Comment:
When I tried to run this version of the action, it failed with `Error: Unhandled error: TypeError: comments.reverse is not a function`. I think it happens because `github.rest.issues.listComments` needs to be awaited. It's also possible that the actual comments resides in `comments.data`.

Reply:
Ah great catch. I copied it from another implementation I did so I'll double check the syntax.